packages/fxa-content-server/app/scripts/views/connect_another_device.js

```diff
@@ -33,54 +32,18 @@ class ConnectAnotherDeviceView extends FormView {
   beforeRender() {
     const account = this.getAccount();
 
-    // If a relier has explicitly set the `syncPreferences=false` property
-    // this means user dont want to sync. Don't offer CAD
-    const syncPreference = this.relier.get('syncPreference');
-    if (syncPreference === false) {
+    // To avoid to redirection loops the forceView property might be set
+    if (this.model.get('forceView')) {
       return;
     }
 
-    // Check to see if user is enrolled in the CAD QR code experiment. This
-    // takes precedence over the "regular" sms experiment.
-    return this.getEligibleQrCodeCadGroup(account).then(
-      ({ group, country }) => {
-        if (group) {
-          switch (group) {
-            case 'control': {
-              // Current SMS experience
-              return this.replaceCurrentPageWithSmsScreen(
-                account,
-                country,
-                this._showSuccessMessage()
-              );
-            }
-            case 'treatment-a': {
-              // New updated CAD via QR code experience
-              return this.replaceCurrentPageWithQrCadScreen(account);
-            }
-            case 'treatment-b': {
-              // Current non-SMS experience
-              return;
-            }
-          }
-        }
+    if (this.isSignIn() && this._areSmsRequirementsMet(account)) {
```

Comment:
Hi @vbudhram,

We recently updated China FxA from train-181 to train-192, and some users are shown this CAD via QR screen, with the #use-sms-link in it, even without `CN` defined in [country-telephone-info.js](https://github.com/mozilla/fxa/blob/v1.192.0/packages/fxa-content-server/app/scripts/lib/country-telephone-info.js).

It looks like checking `_areSmsRequirementsMet` instead of `_isEligibleForSms` (part of original `getEligibleQrCodeCadGroup`) here might be the culprit?